### DISCUSSION TITLE
Error 413 When Attaching PDF

### DISCUSSION THREAD

**Discussion by JohnnyDevCraft** [requarks/wiki#6925] [Help / Questions]
I am trying to attach a PDF to one of my WIKI Articles, and I receive an Error 413 Content too Large when I try to upload the file.

The file is 1.2 MB in size, and I have altered the master.js to allow up to 10mb as described in other threads.  Any further idea's?

I'm using Azure Storage for file storage.
![2024-01-09_13-26-10](https://github.com/requarks/wiki/assets/574071/ad1e339e-83ed-45b3-b05b-bcc614146075)

**Comment by fnafox** [comment]
There's two things that you may run into when uploading larger files:
1. Wiki.js configuration. You need to adjust the value in `admin > security > uploads: max upload size`
2. Reverse Proxy configuration: if you're using a reverse proxy, it needs to be configured to allow upload size over a certain size, e.g.: client_max_body_size in nginx.

**Reply by JohnnyDevCraft** [reply]
1. This setting has been set to 5242880 bytes or 5 mb.
2. I am using nginx, and it is set to a 10mb limit.

Any other thoughts?

**Reply by rrg92** [reply]
Check nginx logs if it is receiving all MB sent and passing to wiki.js server.

**Reply by rrg92** [reply]
@JohnnyDevCraft, could you assist in troubleshooting Nginx to verify if it's configured to handle the desired amount of data correctly?

As a further diagnostic step, I recommend directly testing the upload functionality to determine if the issue lies within wiki.js or Nginx. You can do this by bypassing Nginx and attempting an upload directly to your server. Please use the following curl command as a test:

```sh
myFilePath="/tmp/MyLargeFile.txt"
TOKEN="Your_JWT_Token_Here"
WikiUrl="http://localhost:Your_Port_Here"
curl -v -F "mediaUpload=@${myFilePath}" -F 'mediaUpload={"folderId":0}' -H "Authorization: Bearer ${TOKEN}" "${WikiUrl}/u"
```

Please make the following adjustments to the script:
- Replace `myFilePath` with the path to a test file of the size you're troubleshooting.
- Substitute `Your_JWT_Token_Here` with a valid JWT token. You can find this token in your browser's developer tools: navigate to the Application tab, then to Cookies, and locate the "jwt" cookie in the right pane. Copy and paste its value in place of `Your_JWT_Token_Here`.
- Update `WikiUrl` to reflect the local URL your Wiki.js instance is running on, which should match the target URL in your Nginx configuration. For example, if Nginx redirects to `http://localhost:3000`, set `WikiUrl="http://localhost:3000"` (without a trailing slash).

If the upload is successful and you receive an "ok" response, the issue likely resides in the Nginx configuration rather than wiki.js. On the other hand, if you encounter the same error as before, the problem may be related to the wiki.js configuration itself.

Please let us know the outcome of this test so we can assist you further.

My local test example:

![image](https://github.com/requarks/wiki/assets/16887793/fb4cf3d1-a98a-4495-8f76-ca523bb87bcd)

**Comment by StoiaCode** [comment]
So after troubleshooting for a day, tripple checking my Traefik settings etc etc I noticed that it appears the limit I could upload was 5mb.
So I made a grep for "5mb" and see, there it was, in /wiki/server/app/data.yml...
Together with a fat warning not to edit that file. 
Savely ignoring that message I went forth to edit it anyway, and see: Suddenly I could upload the files I needed to. 

Now the question is, was I being dumb, and just missed something else entirely, something where I could have set the bodyParserLimit where its intended? (The master.js was edit long ago and did nothing.)
![grafik](https://github.com/requarks/wiki/assets/23234609/1ef3b6ad-9dd3-45e9-ad58-726b51a34b64)

**Reply by rrg92** [reply]
Instead of manually editing this file, consider placing it in your `config.yml`. The section within the file serves as the default setting for configurations that are not explicitly specified.
